pkg/test/framework/components/apps/kube.go

```diff
@@ -262,73 +259,71 @@ var (
 		},
 	}
 
-	_ components.Apps = &kubeComponent{}
-	_ api.Component   = &kubeComponent{}
-	_ io.Closer       = &kubeComponent{}
+	_ Instance  = &kubeComponent{}
+	_ io.Closer = &kubeComponent{}
 )
 
 func appSelector(serviceName string) string {
 	return fmt.Sprintf("%s=%s", appLabel, serviceName)
 }
 
 type kubeComponent struct {
-	scope       lifecycle.Scope
+	id resource.ID
+
 	deployments []*deployment.Instance
-	apps        []components.App
+	apps        []App
 	env         *kube.Environment
-}
-
-// NewKubeComponent factory function for the component
-func NewKubeComponent() (api.Component, error) {
-	return &kubeComponent{
-		apps:        make([]components.App, 0),
-		deployments: make([]*deployment.Instance, len(deploymentFactories)),
-	}, nil
-}
 
-func (c *kubeComponent) Descriptor() component.Descriptor {
-	return descriptors.Apps
+	namespace namespace.Instance
 }
 
-func (c *kubeComponent) Scope() lifecycle.Scope {
-	return c.scope
-}
+func newKube(ctx resource.Context, _ Config) (Instance, error) {
+	env := ctx.Environment().(*kube.Environment)
+	c := &kubeComponent{
+		apps:        make([]App, 0),
+		deployments: make([]*deployment.Instance, len(deploymentFactories)),
+		env:         env,
+	}
+	c.id = ctx.TrackResource(c)
 
-func (c *kubeComponent) Start(ctx context.Instance, scope lifecycle.Scope) (err error) {
-	c.scope = scope
+	var err error
 
-	env, err := kube.GetEnvironment(ctx)
-	if err != nil {
-		return err
+	// Wait for the pods to transition to running.
+	if c.namespace, err = namespace.New(ctx, "apps", true); err != nil {
```

Comment:
nit: we'll probably likely want to allow the user to specify the namespace in the configuration. But apps should eventually be broken apart eventually anyway so that each app is its own component.